admin-auth/src/main/kotlin/com/epam/drill/admin/auth/service/impl/CacheServiceImpl.kt

```diff
@@ -0,0 +1,26 @@
+package com.epam.drill.admin.auth.service.impl
+
+import com.epam.drill.admin.auth.service.CacheService
+import com.github.benmanes.caffeine.cache.Cache
+import io.ktor.auth.*
+
+/**
+ * Implementation of [CacheService], providing caching functionalities.
+ *
+ * @property cache An instance of Caffeine`s [Cache] that stores key-value pairs, where keys are of type [String] and values are [Principal] objects.
+ */
+class CacheServiceImpl(
+    val cache: Cache<String, Principal>
```

Comment:
Will be better make `cache` private. For test cases, we will be able to obtain a cache provider instance in another way

Reply:
done